### ISSUE TITLE
Microformats markup on spoiler posts is wrong

### ISSUE THREAD

**Issue by aaronpk** [tootsuite/mastodon#7926]
Currently, the `p-name` element encompasses both the preview text as well as the spoiler text. This causes problems with rendering from the Microformats data since the post appears to be an "article" where the name of the article is the full preview and spoiler text.

The best solution is to move the `p-name` class onto the same element as `p-summary` which should provide the desired behavior to consumers even if they don't understand the `p-summary`. This will make the post appear to be an "article" where the preview text is the name of the article.

I believe this needs to be changed in a couple places:

* [_simple_status.html.haml](https://github.com/tootsuite/mastodon/blob/61a90186070395e133ad2f8e959bdf003a8615ca/app/views/stream_entries/_simple_status.html.haml#L16)
* [_detailed_status.html.haml](https://github.com/tootsuite/mastodon/blob/61a90186070395e133ad2f8e959bdf003a8615ca/app/views/stream_entries/_detailed_status.html.haml#L15)

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by aaronpk** [comment]
Here's an example of how a Microformats consumer gets an incorrect `name` value for a Mastodon post: https://aaronparecki.com/2018/07/02/7/

**Comment by nightpool** [comment]
cf https://github.com/tootsuite/mastodon/pull/1063 and https://github.com/tootsuite/mastodon/issues/965

**Comment by aaronpk** [comment]
Another example where a comment that my site received is unable to be displayed well due to the mismatched name/content markup: https://aaronparecki.com/2018/07/05/14/

**Comment by Gargron** [comment]
I read somewhere p-name is no longer necessary?

**Comment by aaronpk** [comment]
A change in the Microformats parsing spec from [a few months ago](https://github.com/microformats/microformats2-parsing/issues/6) reduced the cases where parsers would auto-generate a `name` property if there wasn't one in the original markup. 

Previously, when the `name` was autogenerated, people ended up having to add an explicit `name` property to the HTML in order to avoid weird broken-looking names. Now that this change is approved and implemented in a couple parsers, things are working a lot better. I believe the Ruby parser has not yet implemented this change, but it's in the Python and PHP ones.

Certainly one option is Mastodon could just remove the `p-name` class, since the newer parsers would end up seeing those posts with just the `summary` and `content` properties. 

Since afaik no Microformats consumers have the concept of CW/spoiler posts yet, this does raise the issue of how to provide a good fallback behavior for consumers that don't understand spoiler posts. My thought with moving the `p-name` class to the same element as `p-summary` is that the current consumers will see it as an "article", most likely only showing the name of the article. Here's an example of how my site renders comments that have a name (whether or not there is also summary or content)

https://media.aaronpk.com/Screen-Shot-2018-07-05-16-37-09-zuMtyZtLAD.jpg

Similarly, here's what my site shows when I'm replying to a post that has a name:

https://media.aaronpk.com/Screen-Shot-2018-07-05-16-39-25-TjGK0soBkB.jpg

If the `p-name` class is used only for the CW/spoiler preview text, then my site wouldn't show the full content in either the reply context or as a comment, which seems like the safest fallback behavior.

**Comment by Gargron** [comment]
But what about posts with no CW?

**Comment by aaronpk** [comment]
Posts with no CW would end up with just a `content` property. According to the new parsing rules, those look just like normal content-only posts and sites don't see a `name` property and treat them as normal text/microblog/toot/tweet posts.

**Comment by aaronpk** [comment]
The thing that stopped me from sending a PR for this is that the Ruby parser doesn't have the updated rules, so it's not possible to create a good test case in Mastodon for the changes. Just removing the class right now will break a bunch of tests that are expecting the name property.

**Comment by Gargron** [comment]
Yep I just discovered that. Could just remove those particular tests...

**Comment by aaronpk** [comment]
That would work 😃

The only downside to this change is until the rest of the mf2 parsers are updated they'll be getting bad parse results from Mastodon permalinks. Could be good incentive to hurry up on updating those parsers tho 😂

**Comment by kevinmarks** [comment]
That matches what I did when marking up mastodon posts in mastoview
```
<div class="h-entry entry">
<div class="p-author h-card header">
<img src="https://gadget.inpocket.net/system/accounts/avatars/000/000/583/original/4c8c9e339a386c23.png" class="u-photo">
<a class="p-name" href="https://gadget.inpocket.net/@binetsuman">kk</a> @<span class="p-nickname">binetsuman</span> 
</div>

<details>
<summary class="p-summary">MacPro2009と2008を無料で手に入れたので2009だけ修繕。以下長々。 <span class="showmore">Show More</span></summary>
<div class="e-content content">
<p>MacPro2009はメモリ足したりGPU入れ替えたりSSDにしたりOS最新にさせたり、そして殻割りなしでCPU(x5675)変えたりして現役マシンにアップグレード。<br>当時のXserveも殻割りしたCPUを使ってたので、それに合わせた形でMP2009もCPUのヒートスプレッダを取り外しただけにすぎないと見たので、殻割りなしでヒートシンクとVRMの間に3mm厚のサーマルパッドを挟んで安定させた状態で使用することに。<br>色々省略するけど無事成功。<br>そのあとずっとベンチマークテストや数TBのファイル交換やOS入れ替えとそれに伴う数日間放置(エアコンなし)、レンダリングソフト等高負荷な処理を連続的にさせてきたけど全く不具合なし。<br>AfterEffects上でのベンチに関してはMacMini2012corei7と特に差がなかったのがちょっと残念。<br>RAMの問題かも。<br>MP2008はRAMがメモリライザーカードごと盗難されたのを譲り受けたままなので未だ漬物石。未チェック。</p>
</div>
</details>
<div class="footer"><a class="u-url" href="https://gadget.inpocket.net/@binetsuman/100330336326622860"><time class="dt-published" datetime="2018-07-06T23:24:00.022Z">13 hours ago</time></a></div> 
</div>
```
